src/events/BillsScript.ts

```diff
@@ -32,69 +32,74 @@ function onBillsPageLoad(): void {
 	jQuery(($) => {
 
 		$(document).ready(async function() {
-
+			$();
 			bills = await getBills();
-			bills.sort((a, b) => a.dateCreated > b.dateCreated ? - 1 : Number(a.dateCreated < b.dateCreated));
-
-			await bills.forEach(function(bill) {
-
-				const allPaid = bill.transactionRequests.every((tr) => {
-					if(tr.paid == false) {
-						return false;
-					}
-				});
-
-				const date: Date = new Date(Number(bill.dateCreated));
-				const dateFormatted: string = date.getDay()+"/"+date.getMonth()+"/"+date.getFullYear();
-
-				let element = "<div class='card bg-dark text-light text-center col-12 bill-item mb-3 p-3 rounded-0 border-light' bill-id='"+bill.id+"'>";
-					element += "<div>";
-					element += "<span class='float-left bill-date' data-timestamp='"+bill.dateCreated+"'>"+dateFormatted+"</span>";
-					element += "<span class='float-right'>"+bill.totalXrp+" XRP</span>";
-					element += "</div>";
-					element += "<h5 class='card-title text-center mb-0'>"+bill.description+"</h5>";
-					element += "<hr class='border-light horizontal-divider'>";
-
-				let parts = "<div class='row'>";
-
-				const weights = bill.weights;
-
-				bill.transactionRequests.forEach(function(tr) {
-					const debtor = tr.debtor.username;
-
-					let weight: number;
-
-					weights.forEach(function(w) {
-						if(w.user.username === debtor) {
-							weight = w.weight;
+			
+			if(bills.length === 0) {
+				document.getElementById("loading").innerHTML =  "No bills where found";
+			} else {
+				$("#loading").hide();
+				bills.sort((a, b) => a.dateCreated > b.dateCreated ? - 1 : Number(a.dateCreated < b.dateCreated));
+				await bills.forEach(function(bill) {
+					
+					const date: Date = new Date(Number(bill.dateCreated));
+					const dateFormatted: string = date.getDay()+"/"+date.getMonth()+"/"+date.getFullYear();
+	
+					let element = "<div class='card bg-dark text-light text-center col-12 bill-item mb-3 p-3 rounded-0 border-light' bill-id='"+bill.id+"'>";
+						element += "<div>";
+						element += "<span class='float-left bill-date' data-timestamp='"+bill.dateCreated+"'>"+dateFormatted+"</span>";
+						element += "<span class='float-right'>"+bill.totalXrp+" XRP</span>";
+						element += "</div>";
+						element += "<h5 class='card-title text-center mb-0'>"+bill.description+"</h5>";
+						element += "<hr class='border-light horizontal-divider'>";
+	
+					let parts = "<div class='row'>";
+	
+					const weights = bill.weights;
+	
+					let allPaid = true;
+					bill.transactionRequests.forEach(function(tr) {
```

Comment:
I used every() instead of forEach() since every() stops when it comes across an unpaid bill, and thus no need to iterate over the rest of the participants

Reply:
It wasn't working, the paid bill never showed up as payed, that is why I changed it, that forEach() was already there so I did not add a loop for that functionality.